### PR TITLE
Fix types in paths

### PR DESCRIPTION
**Pull request by artemdemo**
In this PR I'm introducing a solution for the issue: https://github.com/aleclarson/vite-tsconfig-paths/issues/179

### CODE REVIEW

File: test/types-in-paths/types-in-paths.test.ts
Comment (artemdemo) [inline test/types-in-paths/types-in-paths.test.ts:1]: I introduced a new test because it uses a different approach: I'm explicitly checking that the test fails.

I preferred to create a separate test since it's an edge case. In my opinion, it's not worth modifying the main tests to handle this scenario, and it's clearer to keep it in a separate file.

File: vitest.config.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { defineConfig } from 'vitest/config'
 
 export default defineConfig({
   test: {
-    include: ['test/*.test.ts'],
+    include: ['test/**/*.test.ts'],
```
Comment (artemdemo) [inline vitest.config.ts:5]: Update configuration to support tests in nested directories

File: test/test-infra.ts
Comment (artemdemo) [inline test/test-infra.ts:1]: Moved shared test infrastructure logic here.
